# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../log4cpp/gcc41.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

include <memory> for the declaration of auto_ptr.
abs() for integers is not defined in <cmath>, but in <cstdlib>

--- log4cpp-1.0/src/BasicLayout.cpp.vanilla	2008-08-21 23:50:15.000000000 +0200
+++ log4cpp-1.0/src/BasicLayout.cpp	2008-08-21 23:50:37.000000000 +0200
@@ -15,6 +15,8 @@
 #include <sstream>
 #endif
 
+#include <memory>
+
 namespace log4cpp {
 
     BasicLayout::BasicLayout() {
--- log4cpp-1.0/src/PatternLayout.cpp.vanilla	2008-08-21 23:34:28.000000000 +0200
+++ log4cpp-1.0/src/PatternLayout.cpp	2008-08-21 23:34:49.000000000 +0200
@@ -21,6 +21,7 @@
 #include <strstream>
 #endif
 
+#include <cstdlib>
 #include <iomanip>
 #include <ctime>
 #include <cmath>
